### DISCUSSION TITLE
Possible to commit files after workflow runs?

### DISCUSSION THREAD

**Discussion by chris-c-thomas** [community/community#26152] [Actions]
<p>I’m trying to have GitHub perform an action on some files in my repo every time one of them gets edited within the GH web interface.</p>
<p>The action performs but the files that are generated afterwards are lost when the workflow finishes.</p>
<p>I’d like them to go into an ‘output/‘ directory inside the root of my repo. How can I go about this?</p>

**Comment by weide-zhou** [comment, marked as answer]
<p>Hi <a class="mention" href="/chris-c-thomas">@chris-c-thomas</a> ,</p>
<p>Github hosted runner will be released after workflow completed, thus it will not store the generated files, you need to push the files to repo.</p>
<p>Step1: put the generated files into dir ‘output’.</p>
<p>Step2: push the output folder to your repo.</p>
<p>You can check my sample code below, my repo <a href="https://github.com/weide-zhou/ticket13/tree/B77" rel="nofollow noopener">here</a>.</p>
<pre><code>name: commitfiles
on: [push]

jobs:
  job1:
    runs-on: [ubuntu-latest]
    steps:
      - name: checkout
        uses: actions/checkout@v2
      - name: Modify value # Tried to modify the files based on existing testfile.txt
        run: |
          echo 'Add value2' &gt;&gt; testfile.txt
      - name: move to dir # Move the generated files into output folder
        run: |
          mkdir -p output
          yes| cp -rf testfile.txt ./output/
      - name: Commit files # commit the output folder
        run: |
          git config --local user.email "test@github.com"
          git config --local user.name "GitHub Action test"
          git add ./output
          git commit -m "Add changes"
      - name: Push changes # push the output folder to your repo
        uses: ad-m/github-push-action@master
        with:
          branch: B77 #ignore if your branch is master
          github_token: ${{ secrets.GITHUB_TOKEN }}
          force: true
</code></pre>
<p>Hope it helps.</p>

**Comment by chris-c-thomas** [comment]
<p><span class="mention">@weide-zhouthis</span> is exactly what I was looking for! Thank you for providinng an example, it’s clear and concise!</p>
